test/integration/RedisResourceLockerIntegration.test.ts

```diff
@@ -0,0 +1,218 @@
+    });
+    expect(response.status).toBe(205);
+    response = await fetch(fileUrl);
+    expect(response.status).toBe(404);
+  });
+
+  it('can create a folder and delete it.', async(): Promise<void> => {
+    const containerPath = 'secondfolder/';
+    const containerUrl = `${baseUrl}${containerPath}`;
+    // PUT
+    let response = await fetch(containerUrl, {
+      method: 'PUT',
+      headers: {
+        'content-type': 'text/plain',
+      },
+    });
+    expect(response.status).toBe(205);
+
+    // GET
+    response = await fetch(containerUrl);
+    expect(response.status).toBe(200);
+
+    // DELETE
+    response = await fetch(containerUrl, {
+      method: 'DELETE',
+    });
+    expect(response.status).toBe(205);
+    response = await fetch(containerUrl);
+    expect(response.status).toBe(404);
+  });
+
+  it('can upload and delete an image.', async(): Promise<void> => {
+    const fileUrl = `${baseUrl}image.png`;
+    const fileData = await fs.readFile(
+      joinFilePath(__dirname, '../assets/testimage.png'),
+    );
+
+    let response = await fetch(fileUrl, {
+      method: 'PUT',
+      headers: {
+        'content-type': 'image/png',
+      },
+      body: fileData,
+    });
+    expect(response.status).toBe(205);
+
+    // GET
+    response = await fetch(fileUrl);
+    expect(response.status).toBe(200);
+    expect(response.headers.get('content-type')).toBe('image/png');
+
+    // DELETE
+    response = await fetch(fileUrl, {
+      method: 'DELETE',
+    });
+    expect(response.status).toBe(205);
+    response = await fetch(fileUrl);
+    expect(response.status).toBe(404);
+  });
+
+  it('can get a resource multiple times.', async(): Promise<void> => {
+    const fileUrl = `${baseUrl}image.png`;
+    const fileData = await fs.readFile(
+      joinFilePath(__dirname, '../assets/testimage.png'),
+    );
+
+    let response = await fetch(fileUrl, {
+      method: 'PUT',
+      headers: {
+        'content-type': 'image/png',
+      },
+      body: fileData,
+    });
+    expect(response.status).toBe(205);
+
+    // GET
+    response = await fetch(fileUrl);
+    expect(response.status).toBe(200);
+    expect(response.headers.get('content-type')).toBe('image/png');
+    response = await fetch(fileUrl);
+    expect(response.status).toBe(200);
+    expect(response.headers.get('content-type')).toBe('image/png');
+    response = await fetch(fileUrl);
+    expect(response.status).toBe(200);
+    expect(response.headers.get('content-type')).toBe('image/png');
+    response = await fetch(fileUrl);
+    expect(response.status).toBe(200);
+    expect(response.headers.get('content-type')).toBe('image/png');
+
+    // DELETE
+    response = await fetch(fileUrl, {
+      method: 'DELETE',
+    });
+    expect(response.status).toBe(205);
+    response = await fetch(fileUrl);
+    expect(response.status).toBe(404);
+  });
+
+  describe('Test the ResourceLocker itself', (): void => {
+    const identifier = { path: 'http://test.com/foo' };
+
```

Comment:
```suggestion

    it('is not mocked.', async(): Promise<void> => {
      expect((redis.createClient as jest.Mock).mockImplementation).toBeUndefined();
    });
    
```
To make sure the redis import is not mocked for this test since it can't be mocked correctly in the unit test.